app/models/user.rb

```diff
@@ -43,14 +43,9 @@ def by_name
     end
   end
 
-  def self.find_for_authentication(tainted_conditions)
-    if tainted_conditions[:subdomains].present?
-      local_authority = LocalAuthority.find_by(subdomain: tainted_conditions[:subdomains].first)
-      tainted_conditions.delete(:subdomains)
-      find_first_by_auth_conditions(tainted_conditions.merge(local_authority_id: local_authority.id))
-    else
-      find_first_by_auth_conditions(tainted_conditions)
-    end
+  def self.find_for_database_authentication(tainted_conditions)
```

Comment:
The default implementation of `find_for_database_authentication` just calls `find_for_authentication` so there shouldn't be a need to change this.